library/classes/Document.class.php

```diff
@@ -695,6 +845,48 @@ function createDocument(
         return '';
     }
 
+    /**
+     * Retrieves the document data that has been saved to the filesystem or couch db.  If the $decrypt flag is set to
+     * true, it will return the unencrypted version of the data for the document.
+     * @param bool $decrypt True if the document should be decrypted, false otherwise
+     * @throws BadMethodCallException Thrown if the method is called when the document has been marked as deleted or expired
+     * @return false|string Returns false if the data failed to decrypt, or a string if the data decrypts or is unencrypted.
+     */
+    function get_data($decrypt = false)
+    {
+        $storagemethod = $this->get_storagemethod();
+
+        if ($this->has_expired()) {
+            throw new BadMethodCallException("Should not attempt to retrieve data from expired documents");
+        }
+        if ($this->is_deleted()) {
+            throw new BadMethodCallException("Should not attempt to retrieve data from deleted documents");
+        }
+
+        if ($storagemethod === self::STORAGE_METHOD_COUCHDB) {
+            // Taken from ccr/display.php
+            $couch_docid = $this->get_couch_docid();
+            $couch_revid = $this->get_couch_revid();
+            $couch = new CouchDB();
+            $resp = $couch->retrieve_doc($couch_docid);
+            $data = base64_decode($resp->data);
+        } else {
+            $filepath  = $this->get_url_filepath();
+            if (!file_exists($filepath)) {
+                throw new RuntimeException("Saved filepath does not exist at location " . $filepath);
+            }
+            $data = file_get_contents($filepath);
+        }
+
+        if (!empty($data) && $decrypt && $this->is_encrypted()) {
```

Comment:
as above, note this decrypt stuff needs to already be done above in couchdb, so may make sense to migrate the decrypt into the above if/else block

Reply:
yep did this.